.github/workflows/build_test_package.yml

```diff
@@ -2,740 +2,320 @@ name: Build, Test and Package
 
 on:
   workflow_dispatch:
-
   push:
     branches:
       - main
     tags:
-      - '*'
-      - '!*-test'
+      - "*"
+      - "!*-test"
   pull_request:
     types: [opened, synchronize, reopened, ready_for_review]
     branches:
-      - 'main'
-      - 'release/**'
-      - 'feature/**'
+      - "main"
+      - "release/**"
+      - "feature/**"
 
 jobs:
   build:
     runs-on: ubuntu-latest
-    if: github.event_name != 'pull_request_target'
-
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
     steps:
       - uses: actions/checkout@v4
-
       - name: Set up JDK
-        uses: actions/setup-java@v4
+        uses: actions/setup-java@v3
         with:
-          distribution: 'temurin'
-          java-version: '21'
-
-      - name: Setup Gradle
+          distribution: "temurin"
+          java-version: "21"
+      - name: setup Gradle
         uses: gradle/actions/setup-gradle@v3
-
+        with:
+          gradle-version: 8.8
+          cache-disabled: true
+          cache-read-only: ${{ github.ref != 'refs/heads/main' }}
```

Comment:
When I recently updated the gradle use in the actions, I decided to try the default config for the cache and see if we have any issue. And for the gradle version, the action takes care of downloading the correct one based on the repo config so no need to configure that as well.

Since you completely removed the `setup-gradle` task elsewhere I think we can remove it here as well.